magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/crawler/RegistryCrawler.scala

```diff
@@ -25,7 +25,13 @@ class RegistryCrawler(interface: RegistryExternalInterface, indexer: SearchIndex
 
   var lastCrawl: Option[Future[Unit]] = None
 
-  def crawlInProgress: Boolean = lastCrawl.map(!_.isCompleted).getOrElse(false)
+  def crawlInProgress: Boolean = lastCrawl.map(!_.isCompleted).getOrElse(false) match {
+    case false => 
+        lastCrawl = None
```

Comment:
I think it'd be better to fix this bug in crawl() - instead of matching on `lastCrawl` do something like

```scala
   def crawl(): Future[Unit] = {
     if (crawlInProgress) lastCrawl.get else {
         lastCrawl = Some(newCrawl())
     }
   }
```

Reply:
I think that is a better idea~
Have updated the code~ 
Thank you!